### DISCUSSION TITLE
Redirect URL

### DISCUSSION THREAD

**Discussion by tm-8** [Aohzan/hass-polar#2] [General]
Hello,

it only started working as soon as I removed the additional path >>/api/polar_auth<< from the redirect URL.
Seems a bit strange to me...

**Comment by Aohzan** [comment]
yes indeed because HA wait on this URL :thinking: , you just have the domain in polar ?

**Reply by tm-8** [reply]
Yes - only my duck.dns-domain + port has been entered in Polar Application as redirect.

**Reply by Aohzan** [reply]
`https://_you_.duckdns.org:1234/api/polar_auth` idoesn't work but `https://_you_.duckdns.org:1234` yes ?

**Reply by tm-8** [reply]
Yes - I have tried the original "way" with the auth folder in several variations - nothing worked. In the end, I've tried just with the bare URL - and that did it. Though it seems a little strange to me

**Reply by tm-8** [reply]
So - I've tried it again with a different account.
Just set up the client, entered FIRST the URL with "/api/polar_auth" ONLY and saved the entry.
First try: Error response
After that I added the **bare URL** and set it as default
![230222_001](https://user-images.githubusercontent.com/88572976/220665405-71698f41-6e95-48d6-9445-327e4637ddd8.png)

Next I filled Client ID / Secret with the URL

![230222_002](https://user-images.githubusercontent.com/88572976/220666656-fd377a63-d325-43aa-820c-2df00880039f.png)

It didn't work at the first attempt, but it worked in the end.

Sorry, but I don't understand why :)

**Reply by tm-8** [reply]
It's not really a workaround and I'm still confused: Just set the default on the api-path-address again and access was denied. After turning back on the simple URL, it worked again...

Next try I've just did: I've deleted the api-path-address and kept the standard URL only --> result: denial
So I've entered the /api/polar_auth again, but kept the standard-URL as default (like in the image above) --> it worked again! :)

**I think the solution is to enter both address variants, but set the standard as default. For whatever reason...**

**Reply by Aohzan** [reply]
Ok thank you, if it happens to someone else, I will add it to the README

**Reply by Aohzan** [reply]
hello @tm-8 
I think I found the problem, can you try with the last release with only the url with the callback in it, it should work

**Comment by tm-8** [comment]
Hello Aohzan,

sorry, forgot to test it - I can confirm, it works now fine and correctly!

Am Do., 2. März 2023 um 11:09 Uhr schrieb Matthieu ***@***.***
